UI/static/js/index.js

```diff
@@ -0,0 +1,180 @@
+    var password = document.forms.user.password;
+    var confirm_password = document.forms.user.cpassword;
+    var idNo = document.forms.user.idNo;
+    var passport_url = document.forms.user.passport_url;
+    var phone_number = document.forms.user.phone_number;
+    
+
+    if (!name_pattern.test(fname.value)){
+        fname.focus();
+        alert("Invalid first name.\nFirst name should only contain alpabets.\nFirst name should not be less than three characters.");
+        return false;
+    }
+
+    if (!name_pattern.test(lname.value)){
+        lname.focus();
+        alert("Invalid last name.\nLast name should only contain alpabets.\nLast name should not be less than three characters.");
+        return false;
+    }
+
+    if (mname.value != ""){
+        if (!name_pattern.test(mname.value)){
+            mname.focus();
+            alert("Invalid middle name.\nMiddle name should only contain alpabets.\nMiddle name should not be less than three characters.");
+            return false;
+        }
+    }
+
+    if (!username_pattern.test(username.value)){
+        username.focus();
+        alert("Invalid username.\nUsername can only contain alpabets, numbers and underscore.\nUsername name should not be less than three characters.");
+        return false;
+    }
+
+    if (!email_pattern.test(email.value)){
+        email.focus();
+        alert("Invalid email.\nPlease provide a valid email");
+        return false;
+    }
+
+    if (!idNo_pattern.test(idNo.value)){
+        idNo.focus();
+        alert("Invalid ID number.\nPlease provide a valid ID Number");
+        return false;
+    }
+
+    if (!image_url_pattern.test(passport_url.value)){
+        passport_url.focus();
+        alert("Invalid passport url.\nPassport url should contain a valid domain and end with either jpg, jpeg or png.");
+        return false;
+    }
+
+    if (!phone_pattern.test(phone_number.value)){
+        phone_number.focus();
+        alert("Invalid phone number.\nPhone number should start with either '2547' or '07' followed by 8 digits. ");
+        return false;
+    }
+
+    if (!passwd_pattern.test(password.value)){
+        password.focus();
+        alert("Invalid password.\nPassword should be more than 8 characters long");
+        return false;
+    }
+
+    if (password.value != confirm_password.value){
+        confirm_password.focus();
+        alert("Passwords do not match!!!");
+        return false;
+    }
+
+    var user_data = {
+        "email": email.value,
+        "firstname": fname.value,
+        "id_no": idNo.value,
+        "lastname": lname.value,
+        "othername": mname.value,
+        "passport_url": passport_url.value,
+        "password": password.value,
+        "phone_number": phone_number.value,
+        "username": username.value
+    };
+
+    console.log(user_data);
+
+    location.replace("dashboard.html");
+
+}
+
+/**
+ * office registration
+ */
+
+function registerOffice(){
```

Comment:
You could have different js files for different pages and functionalities.
Like this could be in `js/pages/offices.js` file
The reason for this is that if you use 1 file, it may end up being bloated with a lot of code that you may find hard to read later on